src/agencies-context.ts

```diff
@@ -0,0 +1,14 @@
+import { createContext } from "preact";
+import { StateUpdater } from "preact/hooks";
+
+interface AgenciesContextValues {
+  agencies: string[];
+  setAgencies: StateUpdater<string[]>;
+}
+
+const AgenciesContext = createContext({
+  agencies: [],
+  setAgencies: () => null,
+} as AgenciesContextValues);
```

Comment:
That we have to pass in `agencies` and `setAgencies` from the rendering component feels a bit odd to me, as opposed to where the context would manage this itself to be self-contained, e.g. with a custom provider component:

```tsx
function AgenciesProvider({ children }) {
  const [agencies, setAgencies] = useState([]);
  return <AgenciesContext.Provider value={{ agencies, setAgencies }}>{children}</AgenciesContext.Provider>;
}
```

That being said, I've found competing advice on this sort of pattern, so not something I feel too strongly about right now.

Reply:
It's also something I was curious to explore for the ReportFilterControlsContext:

https://github.com/18F/identity-reporting/compare/margolis-shuffle-props...aduth-report-filter-controls-context?expand=1

Reply:
Updated in 32c68c59d5c8c526619032ca6cfb9be528d719fc for just the agencies provider